pico/api/src/main/java/io/helidon/pico/api/DependencyInfoBlueprint.java

```diff
@@ -19,30 +19,37 @@
 import java.util.Optional;
 import java.util.Set;
 
-import io.helidon.builder.Builder;
-import io.helidon.builder.Singular;
+import io.helidon.builder.api.Prototype;
+import io.helidon.config.metadata.ConfiguredOption;
 
 /**
  * Aggregates the set of {@link InjectionPointInfo}'s that are dependent upon a specific and common
  * {@link ServiceInfo} definition.
  */
-@Builder
-public interface DependencyInfo {
+@Prototype.Blueprint
+interface DependencyInfoBlueprint {
+    /**
+     * Name of the dependency location, such as a field name, or argument name.
+     *
+     * @return name of the element of this dependency
+     */
+    String elementName();
```

Comment:
This is not right.  There can be many different IP elements (and names) that share the same dependency.